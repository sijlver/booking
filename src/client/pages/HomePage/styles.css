.mainContent {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    padding: 0 3rem;
}

/* ListHotels start */
.listHotels {
    display: flex;
    flex-basis: 70%;
    height: calc(100vh - 11rem);
    overflow: auto;
    margin-right: 1rem;
}

.listHotels > ul {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}

.listHotels::-webkit-scrollbar {
	width: 0.5rem;
	height: 0.5rem;
	background-color: transparent;
}

.listHotels::-webkit-scrollbar-track {
   background-color: transparent;
}

.listHotels::-webkit-scrollbar-thumb {
   border-radius: 1rem;
   background-color: #7f7f7f;
   border: 1px solid transparent;
   background-clip: content-box;
}

.hotelItem {
    position: relative;
    width: 19rem;
    height: 15rem;
    margin: 0 1rem 1rem 0;
}

.hotelItemName {
    position: absolute;
    width: 80%;
    margin: 0;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: white;
    font-size: 1.8rem;
    text-align: center;
    text-decoration: none;
}

.hotelItemCityName {
    font-size: 1.2rem;
    padding: 1rem 0 0 1rem;
    color: white;
}

.hotelItemLink {
    display: block;
    height: 100%;
    text-decoration: none;
    background-color: gray;
    transition: box-shadow .3s;
}

.hotelItemLink:hover {
    box-shadow: 0 0 0 0.1rem gray;
}
/* ListHotels end */

/* FilterPanel start */
.filterPanel {
    flex-basis: 30%;
    margin-right: 10rem;
    background-color: #009688;
    border-radius: 1rem;
    padding: 2rem;
}

.searchField {
    width: 100%;
    border: none;
    line-height: 3rem;
    border-radius: 0.7rem;
    padding: 0 1rem;
    margin-bottom: 2rem;
}

.startDateField, .endDateField {
    width: 45%;
    line-height: 2.5rem;
    border-radius: 0.4rem;
    border: none;
    padding: 0 1rem;
    margin-bottom: 2rem;
}

.startDateField {
    margin-right: 10%;
}

.searchButton {
    font-weight: bold;
    text-transform: uppercase;
    color: white;
    background-color: #0277BD;
    border: none;
    padding: 1rem 1.5rem;
    border-radius: 0.6rem;
    cursor: pointer;
    float: right;
    transition: background-color .3s;
}

.searchButton:hover {
    background-color: #0277bdb8;
}
/* FilterPanel end */
