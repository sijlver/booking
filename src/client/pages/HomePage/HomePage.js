import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import FilterPanel from './components/FilterPanel';
import ListHotels from './components/ListHotels';
import { BusyIndicator } from '../../components';
import { hotelsActions } from '../../actions';
import './styles.css';

function HomePage(props) {
    const { busyIndicator, listHotels, requestListHotels, searchName, dateStart, dateEnd, limit } = props;

    return (
        <div className='mainContent'>
            <FilterPanel
                requestListHotels={requestListHotels}
                searchName={searchName}
                dateStart={dateStart}
                dateEnd={dateEnd}
                limit={limit}
            />
            <ListHotels listHotels={listHotels} />
            <BusyIndicator show={busyIndicator} />
        </div>
    );
}

HomePage.propTypes = {
    listHotels: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
    searchName: PropTypes.string,
    dateStart: PropTypes.string,
    dateEnd: PropTypes.string,
    limit: PropTypes.string,
    busyIndicator: PropTypes.bool,
    requestListHotels: PropTypes.func,
};

const mapStateToProps = state => ({
    searchName: state.hotelsReducer.searchName,
    dateStart: state.hotelsReducer.dateStart,
    dateEnd: state.hotelsReducer.dateEnd,
    limit: state.hotelsReducer.limit,
    listHotels: state.hotelsReducer.listHotels,
    busyIndicator: state.commonReducer.busyIndicator,
});

const mapActionToProps = dispatch => ({
    requestListHotels(city, dateStart, dateEnd, limit) {
        dispatch(hotelsActions.getListHotels(city, dateStart, dateEnd, limit));
    },
});

export default connect(mapStateToProps, mapActionToProps)(HomePage);
