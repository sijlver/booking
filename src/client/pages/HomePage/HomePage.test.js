import React from 'react';
import { HomePage } from './HomePage';
import { shallow  } from "enzyme";
import jest from 'jest-mock';

describe('HomePage', () => {
    it('should render self', () => {
        const props = {
            listHotels: [{ label: 'SelimVseh" Apartment at Shchorsa', key: '1' }],
            searchName: 'searchName',
            dateStart: '10-08-1991',
            dateEnd: '11-12-1991',
            limit: '10',
            busyIndicator: true,
            requestListHotels: jest.fn()
        };
        const wrapper = shallow (
            <HomePage {...props} />
        );

        expect(wrapper).toMatchSnapshot();
    });
});
