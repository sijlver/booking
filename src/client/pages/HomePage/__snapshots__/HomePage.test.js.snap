// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render self 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage
    busyIndicator={true}
    dateEnd="11-12-1991"
    dateStart="10-08-1991"
    limit="10"
    listHotels={
      Array [
        Object {
          "key": "1",
          "label": "SelimVseh\\" Apartment at Shchorsa",
        },
      ]
    }
    requestListHotels={[MockFunction]}
    searchName="searchName"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FilterPanel
          dateEnd="11-12-1991"
          dateStart="10-08-1991"
          limit="10"
          requestListHotels={[MockFunction]}
          searchName="searchName"
        />,
        <ListHotels
          listHotels={
            Array [
              Object {
                "key": "1",
                "label": "SelimVseh\\" Apartment at Shchorsa",
              },
            ]
          }
        />,
        <BusyIndicator
          show={true}
        />,
      ],
      "className": "mainContent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "dateEnd": "11-12-1991",
          "dateStart": "10-08-1991",
          "limit": "10",
          "requestListHotels": [MockFunction],
          "searchName": "searchName",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "listHotels": Array [
            Object {
              "key": "1",
              "label": "SelimVseh\\" Apartment at Shchorsa",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "show": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FilterPanel
            dateEnd="11-12-1991"
            dateStart="10-08-1991"
            limit="10"
            requestListHotels={[MockFunction]}
            searchName="searchName"
          />,
          <ListHotels
            listHotels={
              Array [
                Object {
                  "key": "1",
                  "label": "SelimVseh\\" Apartment at Shchorsa",
                },
              ]
            }
          />,
          <BusyIndicator
            show={true}
          />,
        ],
        "className": "mainContent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dateEnd": "11-12-1991",
            "dateStart": "10-08-1991",
            "limit": "10",
            "requestListHotels": [MockFunction],
            "searchName": "searchName",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "listHotels": Array [
              Object {
                "key": "1",
                "label": "SelimVseh\\" Apartment at Shchorsa",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "show": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
